insights/specs/default.py

```diff
@@ -285,7 +285,7 @@ def cloud_cfg(broker):
                 if network_config:
                     return DatasourceProvider(content=json.dumps(network_config), relative_path=relative_path)
 
-        except OSError:
+        except IOError:
```

Comment:
@xiangce I believe this should be a generic `Exception` instead because we don't want this to raise any exceptions other than `SkipComponent` during collection.  

Also see my concerns about the original parser noted in the Issue #2920.  That should probably be done in another PR since any rules using this parser would need to change, but I'd prefer not to close the issue until that is fixed.

Reply:
@bfahr - I add a safety check before opening the file, please check it again.

And I'm very sorry, I cannot get your concerns, would you please make it a bit clear?

Reply:
I just removed the `fix #2920`, which will keep it open after merging this PR.

Reply:
Just updated the #2920, and It seems that more customers are complaining about such problems.

Reply:
@xiangce not sure which point you are asking for more detail on.  

This should be a bare `Exception` to catch whatever problems may occur with reading the file, not just an `IOError`.

I opened a new Issue #2935 with more detail about the context the datasource is not capturing.

Reply:
@xiangce I discussed with @csams and we came up with the changes in my commit.  Also had to fix some merge conflicts to get this merged this morning.